File: bob_and_k_sequence.py
n, k = map(int, input().split())
seq = list(map(int, input().split()))
y = []
for i in range(n):
    a = seq[i] - k * i
    x = 0
    for j in range(n):
        x += abs((a + k * j) - seq[j])
    y.append(x)
y.sort()
print(y[0])

File: bob_and_k_sequence.js
const [n,k] = readline().split(' ').map(el => parseInt(el));
const a = readline().split(' ').map(el => parseInt(el));

let val = 1000 * 1000 * 1000;

for(let x = -100000; x < 100000; x++) {
	let t = 0;
	for(let i = 0; i < n; i++) {
		t += Math.abs((x + i * k) - a[i]);
	}
	val = Math.min(val, t);
}

console.log(val);
